server/src/middlewares/verifyToken.ts

```diff
@@ -0,0 +1,23 @@
+import {
+  Request, Response, NextFunction,
+} from 'express';
+
+const jwt = require('jsonwebtoken');
+
+const verifyToken = (role:'admin' | 'user') => async (req:Request, res:Response, next:NextFunction) => {
+  try {
+    const { token } = req.cookies;
+    const decoded = await jwt.verify(token, process.env.SECRET_KEY);
```

Comment:
`jwt.verify` working async in case if we give it a callback, in this way it will work synchrounaslly 
So. try to create a function that will return `jwt.verify` as promise then use it here